open_feature/open_feature_client.py

```diff
@@ -152,27 +166,65 @@ def evaluate_flag_details(
         evaluation_context: EvaluationContext = None,
         flag_evaluation_options: typing.Any = None,
     ) -> FlagEvaluationDetails:
+        """
+        Evaluate the flag requested by the user from the clients provider.
+
+        :param flag_type: the type of the flag being returned
+        :param key: the string key of the selected flag
+        :param default_value: backup value returned if no result found by the provider
+        :param evaluation_context: Information for the purposes of flag evaluation
+        :param flag_evaluation_options: Additional flag evaluation information
+        :return: a FlagEvaluationDetails object with the fully evaluated flag from a
+        provider
+        """
+
         if evaluation_context is None:
             evaluation_context = EvaluationContext()
+
+        hook_context = HookContext(
+            flag_key=key,
+            flag_type=flag_type,
+            default_value=default_value,
+            evaluation_context=evaluation_context,
+            client_metadata=None,
+            provider_metadata=None,
+        )
+        merged_hooks = []
+
         try:
-            # The merge below will eventually pull in Evaluation contexts from Hooks
-            invocation_context = EvaluationContext()
+            invocation_context = before_hooks(
+                flag_type, hook_context, merged_hooks, None
```

Comment:
How come we're hardcoding `hints` to `None` - is this something that we need to work on in another PR?

Reply:
Yes hints is something we'll come back to in a later pr